### PR TITLE
Readded zipcode field 

### LINKED ISSUE #2053: Add a Zip Code field to the Patient Model
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**
(Re)add the Zip Code field for Patients

* **What feature or behavior is this required for?**
Chicago needs zip codes to provide accurate funding info! Rather than have them hack it into another field, let's explicitly support Zip Code so we can make sure we're properly clearing it on archive.

* **How could we solve this issue? (Not knowing is okay!)**

We removed this field _way back_ in 2017 as we had no need for it. Issue #1053 and PR #1056. 

Adding it back to the Patient model & the patient info view it was removed from is a great start! Also add it in on the data_entry view: 
https://github.com/DCAFEngineering/dcaf_case_management/blob/bf38f20d99f25942190222acdd24e5ec54360ab7/app/views/patients/data_entry.html.erb#L48-L49

Think about where to add this back into tests, if anywhere!

* **Anything else?**
Non-archiving field, so no need to add to Archive Patient

### PR DESCRIPTION
**Pull request by arjunrawal07**
I rule and have completed some work on Case Manager that's ready for review!

Adds the Patient field 'Zipcode' along with the changes in views, controller, tests.

This pull request makes the following changes:
* Added Zipcode field to Patient Model
* Allows Zipcode param in Patient Controller
* Added Zipcode field to Patient Dashboard View
* Added Zipcode field to Data Entry
* Added new field tests for the above views
* Added English & Spanish i18n translations
* Added Zipcode to dev seed data

It relates to the following issue #s: 
* Fixes #2053

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: test/system/data_entry_test.rb [hunk 1]
```diff
@@ -76,6 +77,7 @@ class DataEntryTest < ApplicationSystemTestCase
         assert has_field? 'City', with: 'Washington'
         assert_equal 'DC', find('#patient_state').value
         assert has_field? 'County', with: 'Wash'
+        asser has_field? 'Zipcode', with: '20009'
```
Comment (lomky) [inline test/system/data_entry_test.rb:80]: Slight typo here!
```suggestion
        assert has_field? 'Zipcode', with: '20009'
```

**Review by lomky — COMMENTED** [review]
This is looking so good! I think we're just a couple steps away from done!

when we touch the i18n files, we need to run this command afterwards and commit the changes:
`i18n-tasks normalize`

**Comment by arjunrawal07** [comment]
@lomky I resolved the conflicts in db/seeds.rb and it's now ready for your review. Thank you!

**Review by lomky — APPROVED** [review]
Looks wonderful! Thanks so much for all the work and seeing this through! :leaves: :shipit:

**Pull request merged**